client/driver/docker.go

```diff
@@ -23,6 +24,28 @@ type DockerDriver struct {
 	fingerprint.StaticFingerprinter
 }
 
+type dockerDriverConfig struct {
+	ImageName     string         `mapstructure:"image"`
+	Command       string         `mapstructure:"command"`
+	Args          string         `mapstructure:"args"`
+	NetworkMode   string         `mapstructure:"network_mode"`
+	PortMap       map[string]int `mapstructure:"port_map"`
+	UserName      string         `mapstructure:"auth.username"`
```

Comment:
Maybe create an dockerDriverAuthConfig and put username, password, email, server address in it